src/server/HandlerServerListener.ts

```diff
@@ -0,0 +1,61 @@
+import type { Server, IncomingMessage, ServerResponse } from 'http';
+import { getLoggerFor } from '../logging/LogUtil';
+import { isError } from '../util/errors/ErrorUtil';
+import { guardStream } from '../util/GuardedStream';
+import type { HttpHandler } from './HttpHandler';
+import { ServerListener } from './ServerListener';
+
+/**
+ * A {@link ServerListener} that attaches an {@link HttpHandler} to the `request` event of a {@link Server}.
+ * All incoming requests will be sent to the provided handler.
+ * Failsafes are added to make sure a valid response is sent in case something goes wrong.
+ *
+ * The `showStackTrace` parameter can be used to add stack traces to error outputs.
+ */
+export class HandlerServerListener extends ServerListener {
+  protected readonly logger = getLoggerFor(this);
+
+  /** The main HttpHandler */
+  private readonly handler: HttpHandler;
+  private readonly showStackTrace: boolean;
+
+  public constructor(handler: HttpHandler, showStackTrace = false) {
+    super();
+    this.handler = handler;
+    this.showStackTrace = showStackTrace;
+  }
+
+  public async handle(server: Server): Promise<void> {
+    server.on('request',
+      async(request: IncomingMessage, response: ServerResponse): Promise<void> => {
+        try {
+          this.logger.info(`Received ${request.method} request for ${request.url}`);
+          const guardedRequest = guardStream(request);
+          guardedRequest.on('error', (error): void => {
```

Comment:
This error handler is recreated so many times, we could just create it in the constructor and reuse.